<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Data preparation, processing and interpretation with R</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="This is the complementary manual for QFAB’s Data preparation, processing and interpretation with R">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="Data preparation, processing and interpretation with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the complementary manual for QFAB’s Data preparation, processing and interpretation with R" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Data preparation, processing and interpretation with R" />
  
  <meta name="twitter:description" content="This is the complementary manual for QFAB’s Data preparation, processing and interpretation with R" />
  

<meta name="author" content="QFAB">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="importing-and-reformatting-data-files.html">
<link rel="next" href="reformat-the-file.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data processing, quality checking & interpretation with R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="how-to-read-this-book.html"><a href="how-to-read-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to read this book</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>1.2</b> Getting started</a></li>
<li class="chapter" data-level="1.3" data-path="a-quick-refresher.html"><a href="a-quick-refresher.html"><i class="fa fa-check"></i><b>1.3</b> A quick refresher</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>2</b> Data Preparation</a><ul>
<li class="chapter" data-level="2.1" data-path="data-lifecycle-management-plan.html"><a href="data-lifecycle-management-plan.html"><i class="fa fa-check"></i><b>2.1</b> Data Lifecycle &amp; Management Plan</a></li>
<li class="chapter" data-level="2.2" data-path="good-data-protocol.html"><a href="good-data-protocol.html"><i class="fa fa-check"></i><b>2.2</b> Good Data Protocol</a></li>
<li class="chapter" data-level="2.3" data-path="further-reading.html"><a href="further-reading.html"><i class="fa fa-check"></i><b>2.3</b> Further Reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="reproducible-reporting-in-r.html"><a href="reproducible-reporting-in-r.html"><i class="fa fa-check"></i><b>3</b> Reproducible reporting in R</a><ul>
<li class="chapter" data-level="3.1" data-path="literate-programming.html"><a href="literate-programming.html"><i class="fa fa-check"></i><b>3.1</b> Literate programming</a></li>
<li class="chapter" data-level="3.2" data-path="open-a-new-knitr-document.html"><a href="open-a-new-knitr-document.html"><i class="fa fa-check"></i><b>3.2</b> Open a new knitR document</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="importing-and-reformatting-data-files.html"><a href="importing-and-reformatting-data-files.html"><i class="fa fa-check"></i><b>4</b> Importing and reformatting data files</a><ul>
<li class="chapter" data-level="4.1" data-path="importing-data.html"><a href="importing-data.html"><i class="fa fa-check"></i><b>4.1</b> Importing data</a></li>
<li class="chapter" data-level="4.2" data-path="reformat-the-file.html"><a href="reformat-the-file.html"><i class="fa fa-check"></i><b>4.2</b> Reformat the file</a></li>
<li class="chapter" data-level="4.3" data-path="transform-data.html"><a href="transform-data.html"><i class="fa fa-check"></i><b>4.3</b> Transform data</a></li>
<li class="chapter" data-level="4.4" data-path="normalise-data.html"><a href="normalise-data.html"><i class="fa fa-check"></i><b>4.4</b> Normalise data</a></li>
<li class="chapter" data-level="4.5" data-path="export-data.html"><a href="export-data.html"><i class="fa fa-check"></i><b>4.5</b> Export data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-manipulation-extracting-and-exporting-records.html"><a href="data-manipulation-extracting-and-exporting-records.html"><i class="fa fa-check"></i><b>5</b> Data manipulation - extracting and exporting records</a><ul>
<li class="chapter" data-level="5.1" data-path="removing-samples.html"><a href="removing-samples.html"><i class="fa fa-check"></i><b>5.1</b> Removing samples</a></li>
<li class="chapter" data-level="5.2" data-path="remove-uninformative-measurements.html"><a href="remove-uninformative-measurements.html"><i class="fa fa-check"></i><b>5.2</b> Remove uninformative measurements</a></li>
<li class="chapter" data-level="5.3" data-path="extracting-specific-measurements.html"><a href="extracting-specific-measurements.html"><i class="fa fa-check"></i><b>5.3</b> Extracting specific measurements</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="integrating-datasets.html"><a href="integrating-datasets.html"><i class="fa fa-check"></i><b>6</b> Integrating datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="combining-two-files-with-identical-row-identifiers.html"><a href="combining-two-files-with-identical-row-identifiers.html"><i class="fa fa-check"></i><b>6.1</b> Combining two files with identical row identifiers</a></li>
<li class="chapter" data-level="6.2" data-path="combining-two-files-with-rows-in-common.html"><a href="combining-two-files-with-rows-in-common.html"><i class="fa fa-check"></i><b>6.2</b> Combining two files with rows in common</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="charts-and-graphical-output.html"><a href="charts-and-graphical-output.html"><i class="fa fa-check"></i><b>7</b> Charts and graphical output</a><ul>
<li class="chapter" data-level="7.1" data-path="refresher-standard-r-plotting-functions.html"><a href="refresher-standard-r-plotting-functions.html"><i class="fa fa-check"></i><b>7.1</b> Refresher - standard R plotting functions</a><ul>
<li class="chapter" data-level="7.1.1" data-path="refresher-standard-r-plotting-functions.html"><a href="refresher-standard-r-plotting-functions.html#plot-function"><i class="fa fa-check"></i><b>7.1.1</b> Plot() function</a></li>
<li class="chapter" data-level="7.1.2" data-path="refresher-standard-r-plotting-functions.html"><a href="refresher-standard-r-plotting-functions.html#boxplots"><i class="fa fa-check"></i><b>7.1.2</b> Boxplots</a></li>
<li class="chapter" data-level="7.1.3" data-path="refresher-standard-r-plotting-functions.html"><a href="refresher-standard-r-plotting-functions.html#heatmaps"><i class="fa fa-check"></i><b>7.1.3</b> Heatmaps</a></li>
<li class="chapter" data-level="7.1.4" data-path="refresher-standard-r-plotting-functions.html"><a href="refresher-standard-r-plotting-functions.html#pca-plots"><i class="fa fa-check"></i><b>7.1.4</b> PCA plots</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ggplot2.html"><a href="ggplot2.html"><i class="fa fa-check"></i><b>7.2</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="generating-report-with-r.html"><a href="generating-report-with-r.html"><i class="fa fa-check"></i><b>8</b> Generating report with R</a><ul>
<li class="chapter" data-level="8.1" data-path="other-reporting-formats.html"><a href="other-reporting-formats.html"><i class="fa fa-check"></i><b>8.1</b> Other reporting formats</a></li>
<li class="chapter" data-level="8.2" data-path="interactive-reporting-with-shiny.html"><a href="interactive-reporting-with-shiny.html"><i class="fa fa-check"></i><b>8.2</b> Interactive reporting with Shiny</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>9</b> Conclusion</a></li>
<li class="chapter" data-level="" data-path="appendix-exercise-answers.html"><a href="appendix-exercise-answers.html"><i class="fa fa-check"></i>Appendix: Exercise answers</a><ul>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html"><i class="fa fa-check"></i>4 Importing and reformatting data files</a><ul>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html#read-in-the-datafile"><i class="fa fa-check"></i>4.1 Read in the datafile</a></li>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html#reformat-the-file-1"><i class="fa fa-check"></i>4.2 Reformat the file</a></li>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html#transform-data-1"><i class="fa fa-check"></i>4.3 Transform data</a></li>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html#normalise-data-1"><i class="fa fa-check"></i>4.4 Normalise data</a></li>
<li class="chapter" data-level="" data-path="importing-and-reformatting-data-files-1.html"><a href="importing-and-reformatting-data-files-1.html#export-data-1"><i class="fa fa-check"></i>4.5 Export data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation.html"><a href="data-manipulation.html"><i class="fa fa-check"></i>5.1 Data manipulation</a><ul>
<li class="chapter" data-level="" data-path="data-manipulation.html"><a href="data-manipulation.html#removing-samples-1"><i class="fa fa-check"></i>5.1.2 Removing samples</a></li>
<li class="chapter" data-level="" data-path="data-manipulation.html"><a href="data-manipulation.html#remove-uninformative-measurements-1"><i class="fa fa-check"></i>5.1.2 Remove uninformative measurements</a></li>
<li class="chapter" data-level="" data-path="data-manipulation.html"><a href="data-manipulation.html#extracting-specific-measurements-1"><i class="fa fa-check"></i>5.1.3 Extracting specific measurements</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="an-alternative-practice-data-set.html"><a href="an-alternative-practice-data-set.html"><i class="fa fa-check"></i>5.2 An alternative practice data set</a></li>
<li class="chapter" data-level="" data-path="integrating-and-summarising-datasets.html"><a href="integrating-and-summarising-datasets.html"><i class="fa fa-check"></i>6 Integrating and summarising datasets</a><ul>
<li class="chapter" data-level="" data-path="integrating-and-summarising-datasets.html"><a href="integrating-and-summarising-datasets.html#combining-two-files-with-identical-row-identifiers-1"><i class="fa fa-check"></i>6.1.1 Combining two files with identical row identifiers</a></li>
<li class="chapter" data-level="" data-path="integrating-and-summarising-datasets.html"><a href="integrating-and-summarising-datasets.html#combining-two-files-with-rows-in-common-1"><i class="fa fa-check"></i>6.1.2 Combining two files with rows in common</a></li>
<li class="chapter" data-level="" data-path="integrating-and-summarising-datasets.html"><a href="integrating-and-summarising-datasets.html#linking-in-reference-data"><i class="fa fa-check"></i>6.1.3 Linking in reference data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="summarising-data.html"><a href="summarising-data.html"><i class="fa fa-check"></i>6.2 Summarising data</a><ul>
<li class="chapter" data-level="" data-path="summarising-data.html"><a href="summarising-data.html#table-summaries"><i class="fa fa-check"></i>6.2.1 Table summaries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="charts-and-graphical-output-1.html"><a href="charts-and-graphical-output-1.html"><i class="fa fa-check"></i>7 Charts and graphical output</a><ul>
<li class="chapter" data-level="" data-path="charts-and-graphical-output-1.html"><a href="charts-and-graphical-output-1.html#plot"><i class="fa fa-check"></i>7.1.1 Plot</a></li>
<li class="chapter" data-level="" data-path="charts-and-graphical-output-1.html"><a href="charts-and-graphical-output-1.html#further-timecourse-plotting-practice"><i class="fa fa-check"></i>7.1.2 Further timecourse plotting practice</a></li>
<li class="chapter" data-level="" data-path="charts-and-graphical-output-1.html"><a href="charts-and-graphical-output-1.html#heatmaps-1"><i class="fa fa-check"></i>7.1.3 Heatmaps</a></li>
<li class="chapter" data-level="" data-path="charts-and-graphical-output-1.html"><a href="charts-and-graphical-output-1.html#pca-plots-1"><i class="fa fa-check"></i>7.1.4 PCA plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="ggplot2-1.html"><a href="ggplot2-1.html"><i class="fa fa-check"></i>7.2 ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-sessioninfo-for-this-version-of-the-course-material.html"><a href="appendix-sessioninfo-for-this-version-of-the-course-material.html"><i class="fa fa-check"></i>Appendix - SessionInfo for this version of the course material</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data preparation, processing and interpretation with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-data" class="section level2">
<h2><span class="header-section-number">4.1</span> Importing data</h2>
<p>R provides a variety of tools for reading in data from a file. One of the more useful ones for bioinformatics data is the read.table function, which we will use to read in the first test dataset. This file contains microarray gene expression results from the widely-used Golub et al. (1999) paper. This data set contains calls from around 7,000 gene probes on samples from 38 leukemia patients. Twenty seven patients are diagnosed as having acute lymphoblastic leukemia (<span class="math inline">\(ALL\)</span>) and eleven as having acute myeloid leukemia (<span class="math inline">\(AML\)</span>).</p>
<div class="rmdexercise">
<p>
<strong>Read in the datafile</strong>
</p>
<p>
The data file is in the folder <code>/data/Data_interpretation_with_R</code> and is called <strong>data_set_ALL_AML_train.tsv</strong>.
</p>
<p>
<em>Note</em>: Remember the location of the data file uses a <strong>relative path</strong> to find the dataset, so it depends on the current directory from which you are running the Markdown document. If you are running the document from the newly created Data_Workshop folder, then you need to use the relative path <code>../data/Data_Interpretation/data_set_ALL_AML_train.tsv</code> to read the file. This will go up one directory using the <code>..</code> annotation for parent directory.
</p>
<ol style="list-style-type: decimal">
<li>
<p>
Use the <code>head()</code> and <code>readLines()</code> functions to look at the structure of the file
</p>
</li>
<li>
With the read.table function, read the file contents into a data frame
<ul>
<li>
Use the second column (labelled <strong>Gene Accession Number</strong>, actually the microarray probe ID) as the row names
</li>
</ul>
</li>
</ol>
</div>
<pre><code>## [1] &quot;Gene Description\tGene Accession Number\t1\tcall\t2\tcall\t3\tcall\t4\tcall\t5\tcall\t6\tcall\t7\tcall\t8\tcall\t9\tcall\t10\tcall\t11\tcall\t12\tcall\t13\tcall\t14\tcall\t15\tcall\t16\tcall\t17\tcall\t18\tcall\t19\tcall\t20\tcall\t21\tcall\t22\tcall\t23\tcall\t24\tcall\t25\tcall\t26\tcall\t27\tcall\t34\tcall\t35\tcall\t36\tcall\t37\tcall\t38\tcall\t28\tcall\t29\tcall\t30\tcall\t31\tcall\t32\tcall\t33\tcall&quot;
## [2] &quot;AFFX-BioB-5_at (endogenous control)\tAFFX-BioB-5_at\t-214\tA\t-139\tA\t-76\tA\t-135\tA\t-106\tA\t-138\tA\t-72\tA\t-413\tA\t5\tA\t-88\tA\t-165\tA\t-67\tA\t-92\tA\t-113\tA\t-107\tA\t-117\tA\t-476\tA\t-81\tA\t-44\tA\t17\tA\t-144\tA\t-247\tA\t-74\tA\t-120\tA\t-81\tA\t-112\tA\t-273\tA\t-20\tA\t7\tA\t-213\tA\t-25\tA\t-72\tA\t-4\tA\t15\tA\t-318\tA\t-32\tA\t-124\tA\t-135\tA&quot;                                          
## [3] &quot;AFFX-BioB-M_at (endogenous control)\tAFFX-BioB-M_at\t-153\tA\t-73\tA\t-49\tA\t-114\tA\t-125\tA\t-85\tA\t-144\tA\t-260\tA\t-127\tA\t-105\tA\t-155\tA\t-93\tA\t-119\tA\t-147\tA\t-72\tA\t-219\tA\t-213\tA\t-150\tA\t-51\tA\t-229\tA\t-199\tA\t-90\tA\t-321\tA\t-263\tA\t-150\tA\t-233\tA\t-327\tA\t-207\tA\t-100\tA\t-252\tA\t-20\tA\t-139\tA\t-116\tA\t-114\tA\t-192\tA\t-49\tA\t-79\tA\t-186\tA&quot;                           
## [4] &quot;AFFX-BioB-3_at (endogenous control)\tAFFX-BioB-3_at\t-58\tA\t-1\tA\t-307\tA\t265\tA\t-76\tA\t215\tA\t238\tA\t7\tA\t106\tA\t42\tA\t-71\tA\t84\tA\t-31\tA\t-118\tA\t-126\tA\t-50\tA\t-18\tA\t-119\tA\t100\tA\t79\tA\t-157\tA\t-168\tA\t-11\tA\t-114\tA\t-85\tA\t-78\tA\t-76\tA\t-50\tA\t-57\tA\t136\tA\t124\tA\t-1\tA\t-125\tA\t2\tA\t-95\tA\t49\tA\t-37\tA\t-70\tA&quot;                                                         
## [5] &quot;AFFX-BioC-5_at (endogenous control)\tAFFX-BioC-5_at\t88\tA\t283\tA\t309\tA\t12\tA\t168\tA\t71\tA\t55\tA\t-2\tA\t268\tA\t219\tM\t82\tA\t25\tA\t173\tA\t243\tM\t149\tA\t257\tA\t301\tA\t78\tA\t207\tA\t218\tA\t132\tA\t-24\tA\t-36\tA\t255\tA\t316\tA\t54\tA\t81\tA\t101\tA\t132\tA\t318\tA\t325\tA\t392\tP\t241\tA\t193\tA\t312\tA\t230\tP\t330\tA\t337\tA&quot;                                                                 
## [6] &quot;AFFX-BioC-3_at (endogenous control)\tAFFX-BioC-3_at\t-295\tA\t-264\tA\t-376\tA\t-419\tA\t-230\tA\t-272\tA\t-399\tA\t-541\tA\t-210\tA\t-178\tA\t-163\tA\t-179\tA\t-233\tA\t-127\tA\t-205\tA\t-218\tA\t-403\tA\t-152\tA\t-146\tA\t-262\tA\t-151\tA\t-308\tA\t-317\tA\t-342\tA\t-418\tA\t-244\tA\t-439\tA\t-369\tA\t-377\tA\t-209\tA\t-396\tA\t-324\tA\t-191\tA\t-51\tA\t-139\tA\t-367\tA\t-188\tA\t-407\tA&quot;</code></pre>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="importing-and-reformatting-data-files.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reformat-the-file.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": false,
"twitter": false,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": {}
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["data-proc-qc-inter.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
